### Eclipse Workspace Patch 1.0
#P magnolia-ui-vaadin-widgetset
Index: src/main/java/com/vaadin/terminal/gwt/client/ui/VScrollTable.java
===================================================================
--- src/main/java/com/vaadin/terminal/gwt/client/ui/VScrollTable.java	(revision 45031)
+++ src/main/java/com/vaadin/terminal/gwt/client/ui/VScrollTable.java	(working copy)
@@ -4361,7 +4361,9 @@
                         }
                         if (widget != null) {
                             // widget is now the closest widget to this row
-                            if (widget instanceof VLabel
+                            if ((widget instanceof TableCellWidget && ((TableCellWidget) widget)
+                                    .isRowSelector())
+                                    || widget instanceof VLabel
                                     || widget instanceof VEmbedded
                                     || (widget instanceof VTextField && ((VTextField) widget)
                                             .isReadOnly())) {
